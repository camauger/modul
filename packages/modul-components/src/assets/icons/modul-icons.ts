
export enum ModulIconsCategory {
    Arrow = 'arrow',
    Audio = 'audio',
    Control = 'control',
    Document = 'document',
    File = 'file',
    Form = 'form',
    Image = 'image',
    Video = 'video',
    Sate = 'state'
}

export interface ModulIcon {
    name: string;
    category?: ModulIconsCategory[];
    tag?: string[];
}

export const ModulIcons: ModulIcon[] = [
    {
        name: 'add-circle-filled',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'add-circle',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow--down',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow--left',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow--right',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow--up',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-button--down',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-button--left',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-button--right',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-button--up',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-head-circle--right',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-head-filled--down',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-head-filled--left',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-head-filled--right',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-head-filled--up',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-return',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-thin--down',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-thin--left',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-thin--right',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'arrow-thin--up',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'attachment',
        category: [
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'avatar'
    },
    {
        name: 'calendar',
        category: [
            ModulIconsCategory.Form
        ]
    },
    {
        name: 'chevron--down',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'chevron--left',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'chevron--right',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'chevron--up',
        category: [
            ModulIconsCategory.Arrow,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'clock',
        category: [
            ModulIconsCategory.Form
        ]
    },
    {
        name: 'close-clear',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'code-tag'
    },
    {
        name: 'completed-filled',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'confirmation-white-filled',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'confirmation',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'delete',
        category: [
            ModulIconsCategory.Form,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'download',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'drag-veryical',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'dragndrop',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'edit',
        category: [
            ModulIconsCategory.Form,
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'email-send',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'email'
    },
    {
        name: 'error-filled',
        category: [
            ModulIconsCategory.Form,
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'error',
        category: [
            ModulIconsCategory.Form,
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'evaluation'
    },
    {
        name: 'file-access',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-audio',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'file-code',
        category: [
            ModulIconsCategory.File
        ]
    },
    {
        name: 'file-default',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-dwg',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-excel',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-flash',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document,
            ModulIconsCategory.Video,
            ModulIconsCategory.Image,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'file-image',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document,
            ModulIconsCategory.Image
        ]
    },
    {
        name: 'file-markup',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-mediaplayer',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document,
            ModulIconsCategory.Video,
            ModulIconsCategory.Image,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'file-openoffice-base',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-openoffice-calc',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-openoffice-default',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-openoffice-draw',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-openoffice-impress',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-openoffice-math',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-openoffice-writter',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-pdf',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document,
            ModulIconsCategory.Image
        ]
    },
    {
        name: 'file-powerpoint',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document,
            ModulIconsCategory.Image,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'file-quicktime',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document,
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'file-realplayer',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document,
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'file-text',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-video',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document,
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'file-visio',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-word',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'file-zip',
        category: [
            ModulIconsCategory.File,
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'folder-open',
        category: [
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'folder',
        category: [
            ModulIconsCategory.Document
        ]
    },
    {
        name: 'full-screen',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'grid'
    },
    {
        name: 'grouping',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'heraldry'
    },
    {
        name: 'hide',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'hint'
    },
    {
        name: 'home',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'image-slideshow',
        category: [
            ModulIconsCategory.Image
        ]
    },
    {
        name: 'image-square',
        category: [
            ModulIconsCategory.Image
        ]
    },
    {
        name: 'image1-filled',
        category: [
            ModulIconsCategory.Image
        ]
    },
    {
        name: 'image2-filled',
        category: [
            ModulIconsCategory.Image
        ]
    },
    {
        name: 'information-filled',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'information-white-filled',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'information',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'leave-full-screen',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'list-filled'
    },
    {
        name: 'list'
    },
    {
        name: 'lock'
    },
    {
        name: 'logout',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'multimedia-external',
        category: [
            ModulIconsCategory.Image,
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'multimedia-filled',
        category: [
            ModulIconsCategory.Image,
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'multimedia',
        category: [
            ModulIconsCategory.Image,
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'not-allowed',
        category: [
            ModulIconsCategory.Control,
            ModulIconsCategory.Form
        ]
    },
    {
        name: 'notification'
    },
    {
        name: 'options',
        category: [
            ModulIconsCategory.Control,
            ModulIconsCategory.Form
        ]
    },
    {
        name: 'panel',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'profile'
    },
    {
        name: 'remove-assignation',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'remove-circle-filled',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'remove-circle',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'replace',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'right-answer',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'role-assignation',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'search',
        category: [
            ModulIconsCategory.Control,
            ModulIconsCategory.Form
        ]
    },
    {
        name: 'show',
        category: [
            ModulIconsCategory.Control
        ]
    },
    {
        name: 'subtitle'
    },
    {
        name: 'text1'
    },
    {
        name: 'text2'
    },
    {
        name: 'video-filled',
        category: [
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'video-slideshow',
        category: [
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'video',
        category: [
            ModulIconsCategory.Video,
            ModulIconsCategory.Audio
        ]
    },
    {
        name: 'warning-filled',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'warning-white-filled',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'warning',
        category: [
            ModulIconsCategory.Sate
        ]
    },
    {
        name: 'wrong-answer',
        category: [
            ModulIconsCategory.Sate
        ]
    }
]
