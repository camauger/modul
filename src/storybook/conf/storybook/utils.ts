export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';

