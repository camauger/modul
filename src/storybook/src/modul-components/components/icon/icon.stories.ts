import { actions } from '@storybook/addon-actions';
import { boolean, object, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import { ModulIcon, ModulIcons } from '@ulaval/modul-components/dist/assets/icons/modul-icons';
import { ICON_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

const ICONLIST_SHORTHAND: string[] = ModulIcons.map((m: ModulIcon) => {
    return m.name;
});

const ICONLIST_LONGHAND: string[] = ModulIcons.map((m: ModulIcon) => {
    return `m-svg__${m.name}`;
});

const getPropExampleTemplate: (html: string, propName: string) => string = (html: string, propName: string) => `<div style="display: flex; align-items: center;">
    ${html}
    <span style="margin-left: 8px">Prop ${propName}: <strong style="color:blue">{{ ${propName} }}</span></strong>
</div>`;

storiesOf(`${modulComponentsHierarchyRootSeparator}${ICON_NAME}`, module)
    .add('sandbox', () => ({
        props: {
            name: {
                default: select('Prop name', ICONLIST_SHORTHAND, 'error')
            },
            svgTitle: {
                default: text('Prop svg-title', '')
            },
            showNameAsClass: {
                default: boolean('Prop show-name-as-class', true)
            },
            size: {
                default: text('Prop size', '30px')
            },
            useSvgSprite: {
                default: boolean('Prop use-svg-sprite', true)
            },
            showCustomSvg: {
                default: boolean('Show custom svg', false)
            }
        },
        computed: {
            customSvg(): string {
                const _this: any = this;
                return _this.showCustomSvg ? require('./custom-icon.svg') : '';
            }
        },
        methods: actions('emitClick', 'emitKeydown', 'emitMouseover', 'emitMouseleave'),
        template: `<${ICON_NAME}
            :name="name"
            :svg-title="svgTitle"
            :show-name-as-class="showNameAsClass"
            :size="size"
            :use-svg-sprite="useSvgSprite"
            :custom-svg="customSvg"
            @click="emitClick"
            @keydown="emitKeydown"
            @mouseover="emitMouseover"
            @mouseleave="emitMouseleave"
        />`
    }))
    .add('name (short)', () => ({
        props: {
            name: {
                default: select('Prop name', ICONLIST_SHORTHAND, 'profile')
            }
        },
        template: getPropExampleTemplate(`<${ICON_NAME} :name="name" />`, 'name')
    }))
    .add('name (long)', () => ({
        props: {
            name: {
                default: select('Prop name', ICONLIST_LONGHAND, 'm-svg__profile')
            }
        },
        template: getPropExampleTemplate(`<${ICON_NAME} :name="name" />`, 'name')
    }))
    .add('svgTitle', () => ({
        props: {
            svgTitle: {
                default: text('Prop svg-title', 'Title1')
            }
        },
        template: getPropExampleTemplate(`<${ICON_NAME} :svg-title="svgTitle" name="profile" />`, 'svgTitle')
    }))
    .add('size', () => ({
        props: {
            size: {
                default: text('Prop size', '30px')
            }
        },
        template: getPropExampleTemplate(`<${ICON_NAME} :size="size" name="profile" />`, 'size')
    }))
    .add('showNameAsClass', () => ({
        props: {
            name: {
                default: select('icon name as class', ICONLIST_SHORTHAND, 'profile')
            }
        },
        template: getPropExampleTemplate(`<${ICON_NAME} :show-name-as-class="true" name="profile" />`, 'name')
    }))
    .add('Custom svg', () => ({
        props: {
            name: {
                default: select('Prop name', ICONLIST_SHORTHAND, 'error')
            },
            useSvgSprite: {
                default: boolean('Prop use-svg-sprite', false)
            },
            showCustomSvg: {
                default: boolean('Show custom svg', true)
            }
        },
        computed: {
            customSvg(): string {
                const _this: any = this;
                return _this.showCustomSvg ? require('./custom-icon.svg') : '';
            }
        },
        template: `<div>
            <${ICON_NAME}
                :name="name"
                :use-svg-sprite="useSvgSprite"
                :custom-svg="customSvg"
                size="4em"
            />
            <p>Prop use-svg-sprite needs to be false if you want to see custom-svg</p>
        </div>`
    }))
    .add('UseSvgSprite (false)', () => ({
        props: {
            name: {
                default: select('Prop name', ICONLIST_SHORTHAND, 'profile')
            },
            useSvgSprite: {
                default: boolean('Prop use-svg-sprite', false)
            }
        },
        template: `<${ICON_NAME}
            :name="name"
            size="28px"
            :use-svg-sprite="useSvgSprite"
        />`
    }));

storiesOf(`${modulComponentsHierarchyRootSeparator}${ICON_NAME}/badge`, module)
    .add('state', () => ({
        props: {
            badge: {
                default: object("{ state: 'completed|error|warning' }", { state: 'completed' })
            },
            size: {
                default: text('Prop size (px)', '30px')
            },
            useSvgSprite: {
                default: boolean('Prop use-svg-sprite', true)
            }
        },
        template: `<${ICON_NAME} :size="size" v-m-badge="badge" name="file-pdf" :use-svg-sprite="useSvgSprite" />`
    }))
    .add('offsetX', () => ({
        props: {
            badge: {
                default: object("{ state: 'completed|error|warning', offsetX: 'valueX' }", {
                    state: 'completed',
                    offsetX: '9px'
                })
            }
        },
        template: `<${ICON_NAME} size="30px" v-m-badge="badge" name="profile"/>`
    }))
    .add('offsetY', () => ({
        props: {
            badge: {
                default: object("{ state: 'completed|error|warning', offsetY: 'valueY' }", {
                    state: 'completed',
                    offsetY: '7px'
                })
            }
        },
        template: `<${ICON_NAME} size="30px" v-m-badge="badge" name="profile" />`
    }))
    .add('state, offsetY, offsetX', () => ({
        props: {
            badge: {
                default: object("{ state: 'completed|error|warning', offsetX: 'valueX', offsetY: 'valueY' }",
                    { state: 'completed', offsetX: '-5px', offsetY: '-8px' })
            },
            size: {
                default: text('size (px)', '30')
            }
        },
        template: `<${ICON_NAME} :size="size" v-m-badge="badge" name="profile" />`
    }));
