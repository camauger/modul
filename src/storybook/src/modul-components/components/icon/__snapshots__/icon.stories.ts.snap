// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-icon Custom svg 1`] = `
<div>
  <!---->
   
  <p>
    Prop use-svg-sprite needs to be false if you want to see custom-svg
  </p>
</div>
`;

exports[`Storyshots modul-components|m-icon UseSvgSprite (false) 1`] = `<!---->`;

exports[`Storyshots modul-components|m-icon name (long) 1`] = `
<div
  style="display: flex; align-items: center;"
>
  <svg
    aria-hidden="true"
    class="m-icon"
    height="20px"
    width="20px"
  >
    <!---->
     
    <use
      aria-hidden="true"
    />
  </svg>
   
  <span
    style="margin-left: 8px;"
  >
    Prop name: 
    <strong
      style="color: blue;"
    >
      m-svg__profile
    </strong>
  </span>
</div>
`;

exports[`Storyshots modul-components|m-icon name (short) 1`] = `
<div
  style="display: flex; align-items: center;"
>
  <svg
    aria-hidden="true"
    class="m-icon"
    height="20px"
    width="20px"
  >
    <!---->
     
    <use
      aria-hidden="true"
    />
  </svg>
   
  <span
    style="margin-left: 8px;"
  >
    Prop name: 
    <strong
      style="color: blue;"
    >
      profile
    </strong>
  </span>
</div>
`;

exports[`Storyshots modul-components|m-icon sandbox 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30px"
  width="30px"
>
  <!---->
   
  <use
    aria-hidden="true"
  />
</svg>
`;

exports[`Storyshots modul-components|m-icon showNameAsClass 1`] = `
<div
  style="display: flex; align-items: center;"
>
  <svg
    aria-hidden="true"
    class="m-icon"
    height="20px"
    width="20px"
  >
    <!---->
     
    <use
      aria-hidden="true"
    />
  </svg>
   
  <span
    style="margin-left: 8px;"
  >
    Prop name: 
    <strong
      style="color: blue;"
    >
      profile
    </strong>
  </span>
</div>
`;

exports[`Storyshots modul-components|m-icon size 1`] = `
<div
  style="display: flex; align-items: center;"
>
  <svg
    aria-hidden="true"
    class="m-icon"
    height="30px"
    width="30px"
  >
    <!---->
     
    <use
      aria-hidden="true"
    />
  </svg>
   
  <span
    style="margin-left: 8px;"
  >
    Prop size: 
    <strong
      style="color: blue;"
    >
      30px
    </strong>
  </span>
</div>
`;

exports[`Storyshots modul-components|m-icon svgTitle 1`] = `
<div
  style="display: flex; align-items: center;"
>
  <svg
    class="m-icon"
    height="20px"
    width="20px"
  >
    <title>
      Title1
    </title>
     
    <use
      aria-hidden="true"
    />
  </svg>
   
  <span
    style="margin-left: 8px;"
  >
    Prop svgTitle: 
    <strong
      style="color: blue;"
    >
      Title1
    </strong>
  </span>
</div>
`;

exports[`Storyshots modul-components|m-icon/badge offsetX 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30px"
  style="overflow: visible;"
  width="30px"
>
  <!---->
   
  <use
    aria-hidden="true"
  />
</svg>
`;

exports[`Storyshots modul-components|m-icon/badge offsetY 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30px"
  style="overflow: visible;"
  width="30px"
>
  <!---->
   
  <use
    aria-hidden="true"
  />
</svg>
`;

exports[`Storyshots modul-components|m-icon/badge state 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30px"
  style="overflow: visible;"
  width="30px"
>
  <!---->
   
  <use
    aria-hidden="true"
  />
</svg>
`;

exports[`Storyshots modul-components|m-icon/badge state, offsetY, offsetX 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30"
  style="overflow: visible;"
  width="30"
>
  <!---->
   
  <use
    aria-hidden="true"
  />
</svg>
`;
